### PR TITLE
[Proposed] Change to support big-endian data operations. See related issue

### PR DESCRIPTION
**Pull request by datasynergyuk**
This change adds big-endian support for data operations. However, it cannot fully work due to to issue #35 

This change is more ambitious that my previous attempts so my apologies if I missed something.

### CODE REVIEW

**Comment by jovibor** [comment]
It doesn't seem to work.
I got the same results in `little` and `big` endians.

image

Here there must be `00 00 00 00 00 00 12 34` instead.

P.S. Also, there are a lots of compiler warnings in Release build.

**Comment by datasynergyuk** [comment]
Ok, I'm sorry you found it didn't work. I will investigate.

I get a couple of compiler warnings but they are the same in stock HexCtrl and my updated version. I assumed they were related to my use of VS2017. I don't get any extra warnings in release build over the stock version. What warnings are you getting please?

File: HexCtrl/src/CHexCtrl.cpp [hunk 1]
```diff
@@ -3438,6 +3439,22 @@ void CHexCtrl::OperData(T* pData, EHexOperMode eMode, T tDataOper, ULONGLONG ull
...
+					*pData = _byteswap_ushort(*pData);
+					break;
+				case 4:
+					*pData = _byteswap_ulong(*pData);
+					break;
+				case 8:
+					*pData = _byteswap_uint64(*pData);
+					break;
+			}
+		}
```
Comment (jovibor) [inline HexCtrl/src/CHexCtrl.cpp:3456]: In that kind of template contexts the `if constexpr` must be used, to totally annihilate runtime overhead.
Like in [this snippet](https://github.com/jovibor/HexCtrl/blob/c214ffef761aa172b215534669341bc944bae704/HexCtrl/src/Dialogs/CHexDlgDataInterp.cpp#L464).

File: HexCtrl/src/CHexCtrl.cpp [hunk 2]
```diff
@@ -3480,6 +3497,22 @@ void CHexCtrl::OperData(T* pData, EHexOperMode eMode, T tDataOper, ULONGLONG ull
...
+				*pData = _byteswap_ushort(*pData);
+				break;
+			case 4:
+				*pData = _byteswap_ulong(*pData);
+				break;
+			case 8:
+				*pData = _byteswap_uint64(*pData);
+				break;
+			}
+		}
```
Comment (jovibor) [inline HexCtrl/src/CHexCtrl.cpp:3515]: Ditto

File: HexCtrl/src/Dialogs/CHexDlgOpers.cpp [hunk 3]
```diff
@@ -92,10 +96,30 @@ BOOL CHexDlgOpers::OnCommand(WPARAM wParam, LPARAM lParam)
...
+			fHere = false;
+		}
+		GetDlgItem(IDC_HEXCTRL_OPERS_RADIO_LE)->EnableWindow(bEnabled);
+		GetDlgItem(IDC_HEXCTRL_OPERS_RADIO_BE)->EnableWindow(bEnabled);
+	}
 	else
 		fHere = false;
 
-	return fHere ? TRUE : CDialogEx::OnCommand(wParam, lParam);
+	return fHere ? true : CDialogEx::OnCommand(wParam, lParam);
```
Comment (jovibor) [inline HexCtrl/src/Dialogs/CHexDlgOpers.cpp:122]: What is this change for?
The function `OnCommand` returns `BOOL` not `bool`, so `TRUE` is fully justified.
The `true` would mean implicit conversion to `int`.

Comment (datasynergyuk) [reply HexCtrl/src/Dialogs/CHexDlgOpers.cpp:122]: That change was developer "auto-pilot". You are correct. I apologise.

**Comment by datasynergyuk** [comment]
Ok. Thank you for the feedback. This change was more ambitious and maybe beyond my skill level. I will try to improve it and offer another PR.

**Comment by jovibor** [comment]
> I will try to improve it and offer another PR.

You can simply add other commits with changes to this PR, no need to open a new one.

**Comment by jovibor** [comment]
Worked-out and closed in 21c1533.

**Pull request closed without merging**